### PR TITLE
Delete S4142 (deprecated)

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_cs.html [hunk 1]
```diff
@@ -1,36 +0,0 @@
-<p><code>if</code> statements with conditions that
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_cs.html:1]: Why is this one deleted? This PR should only delete S4142 related stuff, or am I missing something?

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_vb.html [hunk 2]
```diff
@@ -1,32 +0,0 @@
-<p><code>if</code> statements with conditions that
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_vb.html:1]: Same question for this file.

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Seems to me that 2 files were deleted unintentionally.

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_cs.html (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_cs.html:1]: I've missed these files while deleting S4142 and found them now.

File: analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_vb.html (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Utilities/Rules.Description/S1145_vb.html:1]: See up

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**